peakina/io/s3/s3_utils.py

```diff
@@ -67,6 +67,9 @@ def _s3_open_file_with_retries(fs: s3fs.S3FileSystem, path: str, retries: int) -
             # and we give some time to S3 to settle the file status
             sleep(1)
 
+    logger.error(f"could not open {path}")
+    raise Exception(f"Unable to access the target file : {path}")
```

Comment:
We're losing exception context here. How about just `raising` in case we tried `retries` times ?

Reply:
what is for you the better way to keep the context please ? it's not clear for me, like : 
```python
def _s3_open_file_with_retries(fs: s3fs.S3FileSystem, path: str) -> Any:
    try:
        logger.info(f"opening {path}")
        file = fs.open(path)
        return file
    except Exception as ex:
        logger.warning(f"could not open {path}: {ex}")
        # if the file has just been uploaded, then it might not be visible immediatly
        # but the fail to open has been cached by s3fs
        # so, we invalidate the cache
        fs.invalidate_cache(path)
        # and we give some time to S3 to settle the file status
        sleep(1)

    return None

def s3_open(url: str, *, client_kwargs: dict[str, Any] | None = None) -> IO[bytes]:
    """opens a s3 url and returns a file-like object"""
    access_key, secret, bucketname, objectname = parse_s3_url(url)

    token = None
    if client_kwargs is not None and "session_token" in client_kwargs:
        token = client_kwargs["session_token"]
        client_kwargs = {k: v for k, v in client_kwargs.items() if k != "session_token"} or None

    fs = s3fs.S3FileSystem(key=access_key, secret=secret, client_kwargs=client_kwargs, token=token)

    path = f"{bucketname}/{objectname}"
    ret = tempfile.NamedTemporaryFile(suffix=".s3tmp")
    for _ in range(3):
        if (file := _s3_open_file_with_retries(fs, path)) is not None:
            ret.write(file.read())
            ret.seek(0)
            file.close()

            return ret

    logger.error(f"could not open {path}")
    raise Exception(f"Unable to access the target file : {path}")
```
?